src/ConsoleStyle.fs

```diff
@@ -0,0 +1,282 @@
+            | false -> rows |> Seq.append [header]
+
+        let getColumnWordLength row =
+            row
+            |> Seq.mapi (fun column word -> (column, word |> String.length))
+
+        allRows
+        |> Seq.collect getColumnWordLength
+        |> Seq.fold (fun maxWordLengthsPerColumn (column, wordLength) ->
+            let isColumnMaxLengthStored = maxWordLengthsPerColumn |> Map.containsKey column
+            let isNewColumnLength = not isColumnMaxLengthStored
+            let isCurrentWordLongerThanStored = isColumnMaxLengthStored && wordLength > maxWordLengthsPerColumn.[column]
+
+            match isCurrentWordLongerThanStored || isNewColumnLength with
+            | true -> maxWordLengthsPerColumn |> Map.add column wordLength
+            | false -> maxWordLengthsPerColumn
+        ) Map.empty<int,int>
+        |> Map.toList
+        |> List.map snd
+        |> List.map ((+) 2)    // +1 space for both sides of the word ` word `
+
+    let private formatRow (maxWordLengths: int list) items =
+        items
+        |> Seq.mapi (fun i item ->
+            sprintf " %-*s" (maxWordLengths.[i] - 1) item
+        )
+
+    let private renderTableHeader renderHeaderRow maxWordLengths headerWords =
+        headerWords
+        |> formatRow maxWordLengths
+        |> createRow
+        |> renderHeaderRow
+        maxWordLengths |> renderSeparatorRow
+
+    let private renderTableRows maxWordLengths rows =
+        rows
+        |> Seq.map ((formatRow maxWordLengths) >> createRow)
+        |> Seq.iter (printfn "%s")
+        maxWordLengths |> renderSeparatorRow
+
+    let renderTable renderHeaderRow header rows =
+        let renderTableHeader' = renderTableHeader renderHeaderRow
+        let hasHeaders = header |> isNotEmpty
+        let hasRows = rows |> isNotEmpty
+
+        if hasHeaders || hasRows then
+            let wordMaxLengths = getMaxWordLengthsPerColumn hasHeaders hasRows header rows
+
+            wordMaxLengths |> renderSeparatorRow
+            if hasHeaders then header |> renderTableHeader' wordMaxLengths
+            if hasRows then rows |> renderTableRows wordMaxLengths
+
+[<RequireQualifiedAccess>]
+module Console =
+    open System
+    open System.Drawing
+    open Colorful
+    open ShellProgressBar
+
+    type private Color =
+        | Title
+        | SubTitle
+        | Section
+        | TableHeader
+        | Success
+        | Error
+
+    let private color = function
+        | Title -> Color.Cyan
+        | SubTitle -> Color.Yellow
+        | Section -> Color.Yellow
+        | TableHeader -> Color.DarkGoldenrod
+        | Success -> Color.LimeGreen
+        | Error -> Color.Red
+
+    let private getMaxLengthForOptions options =
+        options
+        |> Seq.map fst
+        |> Seq.maxBy String.length
+        |> String.length
+
+    //
+    // Output style
+    //
+
+    [<CompiledName("Messagef")>]
+    let messagef (format: Printf.TextWriterFormat<('a -> unit)>) (message: 'a): unit =
```

Comment:
isn't 
let messagef = printf
enaugh?

Reply:
yes, I guess it is, but I want to be consistent - since there would still be `Console.message` function, it should have variant with `format` (and yes, it it same as `printfn`)..